packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -169,38 +171,38 @@ export class NumberField extends FormAssociatedNumberField {
     public valueChanged(previousValue, nextValue): void {
         super.valueChanged(previousValue, nextValue);
 
-        const numb = parseFloat(nextValue);
-        let out: number | string = numb == nextValue ? nextValue : numb;
-
-        if (nextValue === "" || isNaN(numb)) {
-            out = "";
-        } else {
-            out = this.getValidValue(numb);
-        }
-
-        this.value = out;
-
-        if (this.proxy instanceof HTMLElement) {
-            this.proxy.value = this.value;
-        }
+        this.updateValue(nextValue);
     }
 
     /**
-     * Ensures that the value is between the min and max values
-     *
-     * @param value - number to evaluate
-     * @returns - a string repesentation
+     * Updates the value. Validates that it's a number, between the min
+     *  and max, updates the proxy and emits events.
      *
+     * @param value - value to be validated
      * @internal
      */
-    private getValidValue(value: number): string {
-        if (this.min !== undefined && value < this.min) {
-            value = this.min;
-        } else if (this.max !== undefined && value > this.max) {
-            value = this.max;
+    private updateValue(value): void {
+        if (value === "" || isNaN(parseFloat(value))) {
+            value = "";
+        } else {
+            value = parseFloat(value);
+            if (this.min !== undefined && value < this.min) {
+                value = this.min;
+            } else if (this.max !== undefined && value > this.max) {
+                value = this.max;
+            }
+
+            value = parseFloat(value.toPrecision(12)).toString();
         }
 
-        return parseFloat(value.toPrecision(12)).toString();
+        if (value != this.value) {
+            this.value = value;
+            if (this.proxy instanceof HTMLElement) {
```

Comment:
Shouldn't this be checking for HTMLInputElement as that has a "value" prop and HTMLElement does not?

Reply:
It should be HTMLInputElement. Updated